File: junit-interface/src/main/java/munit/internal/junitinterface/CustomFingerprint.java
package munit.internal.junitinterface;

import sbt.testing.SubclassFingerprint;

public class CustomFingerprint implements SubclassFingerprint {
  final String suite;
  final String runner;
  final boolean module;

  public CustomFingerprint(String suite, String runner, boolean module) {
    this.suite = suite;
    this.runner = runner;
    this.module = module;
  }

  public static CustomFingerprint of(String suite, String runner) {
    return new CustomFingerprint(suite, runner, false);
  }

  public static CustomFingerprint ofModule(String suite, String runner) {
    return new CustomFingerprint(suite, runner, true);
  }

  @Override
  public boolean isModule() {
    return module;
  }

  @Override
  public String superclassName() {
    return suite;
  }

  @Override
  public boolean requireNoArgConstructor() {
    return true;
  }

  @Override
  public String toString() {
    return "CustomFingerprint{" + "suite='" + suite + '\'' + '}';
  }
}

File: munit/js-native/src/main/scala/munit/internal/junitinterface/CustomFingerprint.scala
package munit.internal.junitinterface

import sbt.testing.SubclassFingerprint

class CustomFingerprint(
    val suite: String,
    _isModule: Boolean
) extends SubclassFingerprint {
  override def isModule(): Boolean = _isModule
  override def superclassName(): String = suite
  override def requireNoArgConstructor(): Boolean = true
}
